examples/tls-client.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: Copyright (c) 2017-2023 slowtec GmbH <post@slowtec.de>
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! Asynchronous TLS client example
+use tokio::net::TcpStream;
+
+use std::{
+    fs::File,
+    io::{self, BufReader},
+    net::SocketAddr,
+    path::Path,
+    sync::Arc,
+};
+
+use pkcs8::der::Decode;
+use rustls_pemfile::{certs, pkcs8_private_keys};
+use tokio_rustls::rustls::{self, Certificate, OwnedTrustAnchor, PrivateKey};
+use tokio_rustls::{webpki, TlsConnector};
+
+fn load_certs(path: &Path) -> io::Result<Vec<Certificate>> {
+    certs(&mut BufReader::new(File::open(path)?))
+        .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid cert"))
+        .map(|mut certs| certs.drain(..).map(Certificate).collect())
+}
+
+fn load_keys(path: &Path, password: Option<&str>) -> io::Result<Vec<PrivateKey>> {
+    let expected_tag = match &password {
+        Some(_) => "ENCRYPTED PRIVATE KEY",
+        None => "PRIVATE KEY",
+    };
+
+    if expected_tag.eq("PRIVATE KEY") {
+        pkcs8_private_keys(&mut BufReader::new(File::open(path)?))
+            .map_err(|_| io::Error::new(io::ErrorKind::InvalidInput, "invalid key"))
+            .map(|mut keys| keys.drain(..).map(PrivateKey).collect())
+    } else {
+        let content = std::fs::read(path).unwrap();
+        let mut iter = pem::parse_many(content)
+            .unwrap()
+            .into_iter()
+            .filter(|x| x.tag == expected_tag)
+            .map(|x| x.contents);
+
+        match iter.next() {
+            Some(key) => match password {
```

Comment:
Why not `match (key, password)`?

Reply:
You neither resolved this issue nor answered my comment. Please do not resolve issues yourself if you simply did nothing and decided to silently ignore it.

Anyway. Then let's leave it as. Though the printed messages do not reflect the actual error as you might have noticed.

Reply:
Sorry, this one i marked by mistake.